src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -38,49 +29,15 @@ export const monoPlatform: Platform = {
     });
   },
 
-  findMethod: findMethod,
-
   callEntryPoint: function callEntryPoint(assemblyName: string) {
     // Instead of using Module.mono_call_assembly_entry_point, we have our own logic for invoking
     // the entrypoint which adds support for async main.
     // Currently we disregard the return value from the entrypoint, whether it's sync or async.
     // In the future, we might want Blazor.start to return a Promise<Promise<value>>, where the
     // outer promise reflects the startup process, and the inner one reflects the possibly-async
     // .NET entrypoint method.
-    const invokeEntrypoint = findMethod('Microsoft.AspNetCore.Blazor', 'Microsoft.AspNetCore.Blazor.Hosting', 'EntrypointInvoker', 'InvokeEntrypoint');
-    this.callMethod(invokeEntrypoint, null, [
-      this.toDotNetString(assemblyName),
-      mono_string_array_new(0) // In the future, we may have a way of supplying arg strings. For now, we always supply an empty string[].
-    ]);
-  },
-
-  callMethod: function callMethod(method: MethodHandle, target: System_Object, args: System_Object[]): System_Object {
-    if (args.length > 4) {
-      // Hopefully this restriction can be eased soon, but for now make it clear what's going on
-      throw new Error(`Currently, MonoPlatform supports passing a maximum of 4 arguments from JS to .NET. You tried to pass ${args.length}.`);
-    }
-
-    const stack = Module.stackSave();
-
-    try {
-      const argsBuffer = Module.stackAlloc(args.length);
-      const exceptionFlagManagedInt = Module.stackAlloc(4);
-      for (let i = 0; i < args.length; ++i) {
-        Module.setValue(argsBuffer + i * 4, args[i], 'i32');
-      }
-      Module.setValue(exceptionFlagManagedInt, 0, 'i32');
-
-      const res = invoke_method(method, target, argsBuffer, exceptionFlagManagedInt);
-
-      if (Module.getValue(exceptionFlagManagedInt, 'i32') !== 0) {
-        // If the exception flag is set, the returned value is exception.ToString()
-        throw new Error(monoPlatform.toJavaScriptString(<System_String>res));
-      }
-
-      return res;
-    } finally {
-      Module.stackRestore(stack);
-    }
+    const invokeEntrypoint = bindStaticMethod('Microsoft.AspNetCore.Blazor', 'Microsoft.AspNetCore.Blazor.Hosting.EntrypointInvoker', 'InvokeEntrypoint');
+    invokeEntrypoint(assemblyName, null);
```

Comment:
Is it really right to pass `null` here as the args? Shouldn't we be passing an empty `string[]`, e.g., the result of calling `mono_string_array_new(0)`?

Reply:
Running in to this issue: https://github.com/mono/mono/issues/18245 if I try passing in an array. I'm passing in a null now and null-coalescing it in `InvokeEntrypoint`. I think we can work out the WASM folks to figure out a nice way to pass arbitrary types here.